### ISSUE TITLE
MaxFlow API

### ISSUE THREAD

**Issue by toyboot4e** [toyboot4e/ac-library-hs#7]
Original ACL makes use of `std::vector`. Shall we follow ACL or separate `Builder` and `MaxFlow`?

**Comment by toyboot4e** [comment]
Growable vector requires another indirection (`MutVar`). I would give it a try and compare the performance.

**Comment by toyboot4e** [comment]
So I've implemented a `GrowVec` version of  max flow (b61e96a). Here's a comparsion in [ACLP - D](https://atcoder.jp/contests/practice2/tasks/practice2_d), though the test case was too small:

`MaxFlow`:

```txt
[INFO] slowest: 0.200039 sec  (for 01-13.txt)
[INFO] max memory: 64.000000 MB  (for 01-09.txt)
```

`MaxFlowGrow`:

```txt
[INFO] slowest: 0.216807 sec  (for 01-13.txt)
[INFO] max memory: 64.000000 MB  (for 01-06.txt)
```

`MaxFlowGrow` is _surely_ slower, but no too much. Now I'm preferring ACL-compatible API over speed.
